metric_learn/base_metric.py

```diff
@@ -369,6 +386,168 @@ def score(self, pairs, y):
     """
     return roc_auc_score(y, self.decision_function(pairs))
 
+  def set_default_threshold(self, pairs, y):
+    """Sets the default threshold on the given dataset.
+
+    Returns a threshold that is the mean between the similar
+    metrics mean and the dissimilar metrics mean.
+
+    See more in the :ref:`User Guide <calibration>`.
+    """
+    similar_threshold = np.mean(self.score_pairs(
+        pairs[(y == 1).ravel()]))
+    dissimilar_threshold = np.mean(self.score_pairs(
+        pairs[(y == -1).ravel()]))
+    self.threshold_ = np.mean([similar_threshold, dissimilar_threshold])
+
+  def set_threshold(self, threshold):
+    """Sets the threshold of the metric learner to the given value `threshold
+
+    See more in the :ref:`User Guide <calibration>`.
+
+    Parameters
+    ----------
+    threshold : float
+      The threshold value we want to set. It's a distance metric with
+      respect to which the predicted distance metric for test pairs will be
+      compared to. If they are superior to the threshold they will be
+      classified as similar (+1), and dissimilar (-1) if not.
+
+    Returns
+    -------
+    self : `_PairsClassifier`
+      The pairs classifier with the new threshold set.
+    """
+    self.threshold_ = threshold
+    return self
+
+  def calibrate_threshold(self, pairs_valid, y_valid, strategy='accuracy',
+                          threshold=None, beta=None):
```

Comment:
threshold is not a good name! we already use threshold for the actual threshold

Reply:
Agreed, what about `min_rate` ?

Reply:
when true negative rate is specified it corresponds the max value desired
so maybe simply `rate`?

Reply:
Or `desired_rate`

Reply:
Why is that? 

When `max_tnr` is specified, we want the max `tnr` for a minimum `tpr` no ?
(and if `max_tpr` is specified we want the max `tpr` for a minimum `tnr`)

Because I think that's what we do currently:
- In the doc we have: 
```
      'max_tpr'
          Selects a decision threshold that yields the highest true positive
          rate with true negative rate at least equal to the value of the
          parameter `min_rate`.
      'max_tnr'
          Selects a decision threshold that yields the highest true negative
          rate with true positive rate at least equal to the value of the
          parameter `min_rate`.
```

- And in the code we have: 
```python
    if strategy == 'max_tpr':
      indices = np.where(1 - fpr >= min_rate)[0]
      max_tpr_index = np.argmax(tpr[indices])
      # note: we want a positive threshold (distance), so we take - threshold
      self.threshold_ = - thresholds[indices[max_tpr_index]]

    if strategy == 'max_tnr':
      indices = np.where(tpr >= min_rate)[0]
      max_tnr_index = np.argmax(1 - fpr[indices])
      # note: we want a positive threshold (distance), so we take - threshold
      self.threshold_ = - thresholds[indices[max_tnr_index]]
```

Reply:
Yes, nevermind! I misread. I think it is somewhat more common to fix the error rate (false negative or false positive) while maximizing the accuracy (true positive or true negative). But it is equivalent to what you are doing, so let's keep it this way